glommio/src/io/buffered_file.rs

```diff
@@ -165,7 +165,7 @@ impl BufferedFile {
             )
         })?;
         Ok(ReadResult::from_sliced_buffer(
-            source.extract_dma_buffer(),
+            source.extract_buffer(),
```

Comment:
the commit message doesn't make sense (commenting here because gh doesn't allow commenting on the commit message).

We already have an `IoBuffer` enum. So what exactly are you introducing? Will read on, but that reads strange

Reply:
Sorry I got confused (I wrote the commit message after the fact). I'll fix it up. Essentially what I meant to say is that we no longer return a `DmaBuffer` or a `Vec` to consumers. Instead we return the IoBuffer directly and it now implements `Deref` so 99% of the time consumers don't even need to match, they have easy access to `&[u8]`.